hy/compiler.py

```diff
@@ -544,6 +548,24 @@ def _parse_lambda_list(self, exprs):
                 args.append(k)
                 ret += self.compile(v)
                 defaults.append(ret.force_expr)
+            elif lambda_keyword == "&kwonly":
+                if not PY3:
+                    raise HyTypeError(expr,
```

Comment:
Yes, but it'll have to be a little later after I find more time to think about how to do it properly. I had originally tried something like this (in native_tests/language.hy)&mdash;
``` hy
(defn test-kwonly []
  "NATIVE: test keyword-only arguments"
  (if PY3
    (do
      ;; [redacted for brevity]
    )
    ;; Python 2
    (let [[attempt-to-use-kwonly
           (try
            (setv this-will-fail (fn [&kwonly foo] (inc foo)))
            (catch [e [Exception]] e))]
          [expected-exception-message
           "keyword-only arguments are only available under Python 3"]]
      (assert (isinstance attempt-to-use-kwonly HyTypeError)
              (.format "expected a HyTypeError, got {}"
                       attempt-to-use-kwonly))
      (assert (= (. attempt-to-use-kwonly args [0])
                 expected-exception-message)
              (.format "expected HyTypeError to have message \"{}\""
                       expected-exception-message)))))
```
but then the entire test run would die on the TypeError ("Ran 1 test in 13.094s / FAILED (errors=1)") instead of catching it like I had intended.